WordPress/Classes/ViewRelated/Gutenberg/Processors/GutenbergMediaFilesUploadProcessor.swift

```diff
@@ -0,0 +1,79 @@
+import Foundation
+import Aztec
+
+class GutenbergMediaFilesUploadProcessor: Processor {
+    private struct FileBlockKeys {
+        static var name = "wp:jetpack/story"
+    }
+
+    let mediaUploadID: Int32
+    let remoteURLString: String
+    let serverMediaID: Int
+
+    init(mediaUploadID: Int32, serverMediaID: Int, remoteURLString: String) {
+        self.mediaUploadID = mediaUploadID
+        self.serverMediaID = serverMediaID
+        self.remoteURLString = remoteURLString
+    }
+
+    lazy var mediaFilesProcessor = GutenbergBlockProcessor(for: FileBlockKeys.name, replacer: { block in
+
+        guard let mediaFileAttributes = block.attributes["mediaFiles"] as? [[String: Any]] else {
+            return nil
+        }
+        let mediaFiles = mediaFileAttributes.map { attributes in
+            return MediaFile(dictionary: attributes)
+        }
+
+        let media: [MediaFile] = mediaFiles.map { mediaFile -> MediaFile in
+            guard Int32(mediaFile.id) == self.mediaUploadID else {
+                return mediaFile
+            }
+
+            let newURL = StoryPoster.filePath.appendingPathComponent("\(self.serverMediaID)")
+
+            do {
+                try FileManager.default.moveItem(at: URL(string: mediaFile.url)!, to: newURL)
```

Comment:
⚠️ Force unwrap (I know that this is one of the case where we should use it but still 😄 )